src/include/ceph_features.h

```diff
@@ -91,6 +91,7 @@ DEFINE_CEPH_FEATURE(15, 1, MONENC)
 DEFINE_CEPH_FEATURE_RETIRED(16, 1, QUERY_T, JEWEL, LUMINOUS)
 DEFINE_CEPH_FEATURE(16, 3, SERVER_O)
 DEFINE_CEPH_FEATURE_RETIRED(17, 1, INDEP_PG_MAP, JEWEL, LUMINOUS)
+DEFINE_CEPH_FEATURE(17, 1, OSD_DELETING_SNAPS)
```

Comment:
we may want to piggy-back this on SERVER_M to avoid spending a distinct bit.

Reply:
I don't think we need the feature bit at all. We only use it when encoding and decoding the pg_pool_t, but we can do that entirely based on the local struct_v. :)

Reply:
What information do we use to choose between 26 and 27 here, if we don't use a flag like OSD_DELETING_SNAPS or SERVER_M?
```c++
  uint8_t v = 27;
  if (!HAVE_FEATURE(features, OSD_DELETING_SNAPS)) {
    v = 26;
  }
```

Reply:
If we go this route I think we do need feature-dependent decoding.  We have a newish requirement that the canonical OSDMaps encoded at the monitor be encoded in the way that the oldest osd in the cluster would encode (to avoid divergence and the need for OSDs to pull full osdmaps from the mon).  And pg_pool_t is in the OSDMap.  We should just reuse SERVER_MIMIC for this.

But.. I'm not so sure we want to go down this path.  I agree that it's a big win compared to the status quo, but once we have a more complete fix I think it will be needlessly complex.  I think the question is whether we expect to address that in the mimic timeframe.  Started pondering that... let's discuss next week?